pandas/core/arrays/interval.py

```diff
@@ -0,0 +1,856 @@
+        if len(closed) != 1:
+            raise ValueError("Intervals must all be closed on the same side.")
+        closed = closed.pop()
+
+        # TODO: avoid intermediate list
+        left = np.concatenate([interval.left for interval in to_concat])
+        right = np.concatenate([interval.right for interval in to_concat])
+        return cls._simple_new(left, right, closed=closed, copy=False)
+
+    def _shallow_copy(self, left=None, right=None):
+        from pandas.core.indexes.interval import IntervalIndex
+
+        if left is None:
+
+            # no values passed
+            # XXX: is ^ right? Or does that mean just left wasn't passed?
+            left, right = self.left, self.right
+
+        elif right is None:
+
+            # only single value passed, could be an IntervalIndex
+            # or array of Intervals
+            if not isinstance(left, (type(self), IntervalIndex)):
+                left = type(self)(left)
+
+            left, right = left.left, left.right
+        else:
+
+            # both left and right are values
+            pass
+
+        return self._simple_new(left, right, closed=self.closed,
+                                verify_integrity=False)
+
+    # TODO: doc
+    def copy(self, deep=False):
+        left = self.left.copy(deep=True) if deep else self.left
+        right = self.right.copy(deep=True) if deep else self.right
+        closed = self.closed
+        # TODO: Could skip verify_integrity here.
+        return type(self).from_arrays(left, right, closed=closed)
+
+    def _formatting_values(self):
+        return self.values
+
+    def get_values(self):
+        return self.values
+
+    def isna(self):
+        return isna(self.left)
+
+    @property
+    def nbytes(self):
+        return self.left.nbytes + self.right.nbytes
+
+    @property
+    def itemsize(self):
+        return self.left.itemsize + self.right.itemsize
+
+    def take(self, indices, axis=0, allow_fill=True, fill_value=None,
+             **kwargs):
+        nv.validate_take(tuple(), kwargs)
+        indices = _ensure_platform_int(indices)
+        left, right = self.left, self.right
+
+        if fill_value is None:
+            fill_value = self._na_value
+        mask = indices == -1
+
+        if not mask.any():
+            # we won't change dtype here in this case
+            # if we don't need
+            allow_fill = False
+
+        taker = lambda x: x.take(indices, allow_fill=allow_fill,
+                                 fill_value=fill_value)
+
+        try:
+            new_left = taker(left)
+            new_right = taker(right)
+        except ValueError:
+
+            # we need to coerce; migth have NA's in an
+            # integer dtype
+            new_left = taker(left.astype(float))
+            new_right = taker(right.astype(float))
+
+        return self._shallow_copy(new_left, new_right)
+
+    take_nd = take
+
+    # Formatting
+
+    def _format_data(self):
```

Comment:
this is really a generic formatter Index. should this be in EA?

Reply:
Or rather somewhere in the formatting code? (and then it could still be called from within a default EA repr)